app/views/layouts/application.haml

```diff
@@ -95,22 +95,22 @@
                 %li= link_to 'Favorites', favorites_path
                 - unless current_user.read_only?
                   %li= link_to "Post", new_post_path
-              = link_to logout_path, method: :delete, id: :"header-logout", data: { confirm: 'Press OK to confirm logout' } do
+              = link_to destroy_user_session_path, method: :delete, id: :"header-logout", data: { confirm: 'Press OK to confirm logout' } do
                 = button_tag "Log out", class: 'button'
-            - elsif !current_page?(login_path)
-              = form_tag login_path, method: :post, id: 'header-form' do
+            - elsif !current_page?(new_user_session_path)
+              = form_with scope: :user, url: user_session_path, id: 'header-form' do |f|
                 #header-forms
-                  = text_field_tag :username, params[:username], placeholder: "Username", id: 'header-username', autocomplete: 'username'
-                  = password_field_tag :password, params[:password], placeholder: "Password", id: 'header-password', autocomplete: 'current-password'
+                  = f.text_field :username, placeholder: "Username", id: 'header-username', autocomplete: 'username'
+                  = f.password_field :password, placeholder: "Password", id: 'header-password', autocomplete: 'current-password'
                 #header-remember
-                  = check_box_tag :remember_me, class: 'vmid', id: 'header-rememberme'
+                  = f.check_box :remember_me, class: 'vmid', id: 'header-rememberme'
                   = label_tag 'header-rememberme', "Remember Me", class: 'vmid'
```

Comment:
I think you need to make the label `f.label` too

Reply:
If I do that, it actually prefixes the `for="..."` bit of the label here, and for needs to match the `id` of the other field. It's a bit weird honestly. I could edit the other to say `id: 'user_header-rememberme'`  and use `f.label`, but they have to be in sync, and Rails adds prefixes like `user` automatically.

Reply:
wow. rails why.